src/bot/commands/rules.command.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Injectable,
+  Logger,
+  UseFilters,
+  UseInterceptors,
+} from '@nestjs/common';
+import {
+  Command,
+  EventParams,
+  Handler,
+  InteractionEvent,
+} from '@discord-nestjs/core';
+import { SlashCommandPipe } from '@discord-nestjs/common';
+import {
+  CommandInteraction,
+  Message,
+  MessagePayload,
+  PermissionFlagsBits,
+} from 'discord.js';
+import { MikroORM, UseRequestContext } from '@mikro-orm/core';
+import { InjectRepository } from '@mikro-orm/nestjs';
+import { EntityRepository } from '@mikro-orm/postgresql';
+
+import { RulesService } from '../rules.service';
+import { TimingLogInterceptor } from '../interceptors/logging.interceptor';
+import { ValidationErrorFilter } from '../filters/validation-error.filter';
+import { CatchallErrorFilter } from '../filters/catchall-error.filter';
+import { GuildEntity } from 'src/entities/guild/guild.entity';
+
+export class CheckScoreDTO {}
+
+@Command({
+  name: 'rules',
+  description: `Rules for Virginity Bot on your server.`,
+  defaultMemberPermissions: PermissionFlagsBits.SendMessages,
+  dmPermission: false,
+})
+@Injectable()
+@UseFilters(ValidationErrorFilter, CatchallErrorFilter)
+@UseInterceptors(TimingLogInterceptor)
+export class RulesCommand {
+  private readonly logger = new Logger(RulesCommand.name);
+
+  constructor(
+    private readonly orm: MikroORM,
+    @InjectRepository(GuildEntity)
+    private readonly guilds: EntityRepository<GuildEntity>,
+    private readonly rulesBoard: RulesService,
+  ) {}
+
+  @Handler()
+  @UseRequestContext()
+  async handler(
+    @InteractionEvent(SlashCommandPipe) dto: CheckScoreDTO,
+    @EventParams() [interaction]: [interaction: CommandInteraction],
+  ): Promise<Message> {
+    if (interaction.guildId == null || interaction.guild == null) {
+      this.logger.error([`interaction.guildId was null somehow`, interaction]);
+      throw new Error(`interaction.guildId was null somehow`);
+    } else if (interaction.channel == null) {
+      this.logger.error([`interaction.channel was null somehow`, interaction]);
+      throw new Error(`interaction.channel was null somehow`);
+    }
+
+    await interaction.deferReply();
```

Comment:
This command should be fast enough that we don't need to defer the reply.

Reply:
It complained at somepoint about it not being deferred. But sure I'll test it again without.

Reply:
Can confirm, event will still fail without a deferral.

Reply:
Did you change the response method? Or just remove the deferral. `.followUp` only works with deferred interactions.